web/src/compose_validate.js

```diff
@@ -341,47 +341,67 @@ function check_unsubscribed_stream_for_send(stream_name, autosubscribe) {
 }
 
 export function wildcard_mention_allowed() {
-    if (
-        page_params.realm_wildcard_mention_policy ===
-        settings_config.wildcard_mention_policy_values.by_everyone.code
-    ) {
-        return true;
-    }
-    if (
-        page_params.realm_wildcard_mention_policy ===
-        settings_config.wildcard_mention_policy_values.nobody.code
-    ) {
-        return false;
-    }
-    if (
-        page_params.realm_wildcard_mention_policy ===
-        settings_config.wildcard_mention_policy_values.by_admins_only.code
-    ) {
-        return page_params.is_admin;
-    }
 
-    if (
-        page_params.realm_wildcard_mention_policy ===
-        settings_config.wildcard_mention_policy_values.by_moderators_only.code
-    ) {
-        return page_params.is_admin || page_params.is_moderator;
-    }
+    // Get the stream_id of the current compose state
+    const stream_id = compose_state.stream_id();
 
-    if (
-        page_params.realm_wildcard_mention_policy ===
-        settings_config.wildcard_mention_policy_values.by_full_members.code
-    ) {
-        if (page_params.is_admin) {
+    // Use the stream_id to get the subscriber count
+    const subscriber_count = peer_data.get_subscriber_count(stream_id) || 0;
+
+    // If subscriber count is greater than the global large stream threshold variable
+    if (subscriber_count > wildcard_mention_large_stream_threshold) {
```

Comment:
To avoid all the re-indentation below, would it make sense to instead invert this conditional? Then we'd only have the 5 lines currently at the bottom of this block indented an extra time.

Reply:
This makes sense. I'll updated accordingly, thanks.